### ISSUE TITLE
Forbid classes with too many public instance attributes

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#691]
# Rule request

## Thesis

Having classes with too many public instance attribute is not great.
It means that this should be either:
1. Refactored as a dataclass (which is explicit DTO)
2. Split into multiple classes
3. Inherited from some other class or mixin

Bad code:

```python
class ClassWithMultipleInstanceAttributes(object):
    def __init__(self):
        self.first = 1
        self.second = 1
        self.third = 1
        self.fourth = 1
        self.fifth = 1
        self.sixth = 1
        self.overflow = 'boom!'  # violation
```

I guess that allowing six element is ok, more than six - should raise a violation.

What is ok?

1. class attributes
2. protected and private instance attributes
3. properties
4. inherited attributes
5. type annotations
6. Dataclasses (since type annotations and class-level attributes are allowed)

## Reasoning

This is a complexity rule.
Complex classes enforce using their mutable attributes in a wrong way.

This rule also should have a setting to be configurable.
